app/bulk_resource/views.py

```diff
@@ -22,33 +32,115 @@ def index():
+            header_row = csv_reader.next()
+            csv_header_row = CsvHeaderRow(
+                csv_header_row_container=csv_container
+            )
+            for column_name in header_row:
+                csv_cell = CsvHeaderCell(data=column_name,
+                                         csv_header_row=csv_header_row)
+                db.session.add(csv_cell)
+            db.session.add(csv_header_row)
 
             # Iterate through the CSV file row-by-row and then cell-by-cell.
             # Each cell contains one comma-separated string in a row of a CSV
             # file.
-            csv_reader = csv.reader(csv_file)
             for row in csv_reader:
-                csv_row = CsvRow(csv_container=csv_container)
+                csv_body_row = CsvBodyRow(csv_container=csv_container)
                 for cell_data in row:
-                    csv_cell = CsvCell(data=cell_data, csv_row=csv_row)
+                    csv_cell = CsvBodyCell(data=cell_data,
+                                           csv_body_row=csv_body_row)
                     db.session.add(csv_cell)
-                db.session.add(csv_row)
+                db.session.add(csv_body_row)
             db.session.add(csv_container)
             db.session.commit()
-
             # TODO: Error catching if CSV is malformed.
-
-        return redirect(url_for('bulk_resource.review'))
+            # TODO: Check that CSV file has "Name", "Address" headings
+        return redirect(url_for('bulk_resource.review1'))
     return render_template('bulk_resource/upload.html', form=form)
 
 
-@bulk_resource.route('/review')
+@bulk_resource.route('/review1', methods=['GET', 'POST'])
 @login_required
-def review():
-    # TODO: Retrieve CSV associated with current_user.
-    csv_container = CsvContainer.query.first()
-    return render_template('bulk_resource/review.html',
-                           csv_container=csv_container)
+def review1():
+    csv_container = CsvContainer.most_recent(user=current_user)
+    if csv_container is None:
+        abort(404)
+    form = DetermineDescriptorTypesForm()
+    if form.validate_on_submit():
+        if form.navigation.data['submit_next']:
+            for i, descriptor_type in enumerate(form.descriptor_types.data):
+                csv_container.csv_header_row.csv_header_cells[i]\
+                    .descriptor_type = descriptor_type
+                db.session.add(
+                    csv_container.csv_header_row.csv_header_cells[i]
+                )
+            db.session.commit()
+            csv_container.predict_options()
+            # TODO: Skip the second review step if there are no option
+            # TODO: descriptor types.
+            return redirect(url_for('bulk_resource.review2'))
+        elif form.navigation.data['submit_back']:
+            db.session.delete(csv_container)
+            return redirect(url_for('bulk_resource.upload'))
+        elif form.navigation.data['submit_cancel']:
```

Comment:
I'd prefer to keep them separate for the moment since I anticipate differing behavior in the future.